File: source/Emphasize/Format Definitions/BoldItalicCode.vb
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition


Namespace FormatDefinitions

    <ClassificationType(ClassificationTypeNames:=BoldItalicCode.Name)>
    <Export(GetType(EditorFormatDefinition))>
    <Name(BoldItalicCode.Name)>
    <Order(After:=Priority.High)>
    <UserVisible(True)>
    Public NotInheritable Class BoldItalicCode
        Inherits ClassificationFormatDefinition


        Public Const Name As String = "Comment - Bold, Italic, Code Span"


        <Export(GetType(ClassificationTypeDefinition))>
        <BaseDefinition("Comment")>
        <Name(Name)>
        Public Shared ReadOnly ClassificationType As ClassificationTypeDefinition


        Public Sub New()
            DisplayName = Name
            IsBold = True
            IsItalic = True
            BackgroundBrush = Code.CreateBackgroundBrush()
            BackgroundOpacity = Code.GetBackgroundOpacity()
        End Sub

    End Class

End Namespace

File: source/Emphasize/FormatDefinitions/BoldItalicCode.cs
#nullable enable

using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;


namespace Emphasize.FormatDefinitions {

    [ClassificationType(ClassificationTypeNames = Name)]
    [Export(typeof(EditorFormatDefinition))]
    [Name(Name)]
    [Order(After = Priority.High)]
    [UserVisible(true)]
    public class BoldItalicCode : ClassificationFormatDefinition {

        public const string Name = "Emphasize - Bold, Italic, Code";


        [Export(typeof(ClassificationTypeDefinition))]
        [Name(Name)]
        public static readonly ClassificationTypeDefinition? ClassificationType;


        public BoldItalicCode() {
            DisplayName = Name;
            IsBold = true;
            IsItalic = true;
            BackgroundBrush = Code.CreateBackgroundBrush();
            BackgroundOpacity = Code.GetBackgroundOpacity();
        }

    }

}
